### PR TITLE
Allow using readOnly property as part of group creation/patching

### PR DESCRIPTION
**Pull request by sarah-pontier**
- Add `readOnly` property to `CreateGroupParams`,  `UpdateGroupParams`,  and `Group` interfaces
 - Update group tests to include readOnly property

### CODE REVIEW

File: packages/saved-views-client/src/models/Group.ts [hunk 1]
```diff
@@ -8,5 +8,6 @@ export interface Group {
   id: string;
   displayName: string;
   shared: boolean;
+  readOnly: boolean;
```
Comment (roluk) [inline packages/saved-views-client/src/models/Group.ts:11]: I'm concerned that a group might not contain `readOnly` property if it was created before this update. Is that plausible? If so, this property should be marked optional.

Comment (sarah-pontier) [reply packages/saved-views-client/src/models/Group.ts:11]: After applying TypeORM migrations in DEV and QA all groups created prior have the readOnly property set to false. 
However in PROD groups do not have the readOnly property, as we haven't released and migrated yet

Comment (roluk) [reply packages/saved-views-client/src/models/Group.ts:11]: By default the client targets PROD environment so I think it's best to wait for the server side change to be propagated first.

File: packages/saved-views-client/src/client/ITwinSavedViewsClient.test.ts [hunk 2]
```diff
@@ -220,6 +220,7 @@ describe("ITwinSavedViewsClient", () => {
       iModelId: "test_imodelid",
```
Comment (roluk) [inline packages/saved-views-client/src/client/ITwinSavedViewsClient.test.ts:220]: Please update [CHANGELOG.md](https://github.com/iTwin/saved-views/blob/sp/readonly-groups/packages/saved-views-client/CHANGELOG.md)

**Pull request merged**